PdCore/src/main/java/org/puredata/android/service/PdService.java

```diff
@@ -241,8 +241,15 @@ private Notification makeNotification(Intent intent, int icon, String title, Str
 				notificationManager.createNotificationChannel(channel);
 			}
 		}
-
-		PendingIntent pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, 0);
+		
+            	PendingIntent pi = null;
+            	if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                	pi = PendingIntent.getActivity(getApplicationContext(), 0, intent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
```

Comment:
I am not sure about setting `FLAG_UPDATE_CURRENT` here. On the one hand, this will just replace an existing notification so it will not show multiple notifications, if `startAudio()` is called multiple times. On the other hand, this should have been prevented in `startAudio()`.

Reply:
What's the solution here? We need to set one or the other per the original discussion on SO.

Reply:
> What's the solution here? We need to set one or the other per the original discussion on SO.

I would pick `FLAG_IMMUTABLE`, but I am unsure about the `FLAG_UPDATE_CURRENT_PART`.